From 3ce1a41144beeee9e8e1840fea8b1d94b388614f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Sebastian=20Dr=C3=B6ge?= <sebastian@centricular.com>
Date: Thu, 2 Oct 2014 10:13:28 +0300
Subject: [PATCH 2/7] capsfilter: Push pending events before a buffer also if
 upstream never configured caps but we have srcpad caps already

Otherwise we never send pending events downstream that arrive after we
configured caps on the srcpad.

https://bugzilla.gnome.org/show_bug.cgi?id=737735
---
 plugins/elements/gstcapsfilter.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/plugins/elements/gstcapsfilter.c b/plugins/elements/gstcapsfilter.c
index 86c0df0..a499453 100644
--- a/plugins/elements/gstcapsfilter.c
+++ b/plugins/elements/gstcapsfilter.c
@@ -340,6 +340,9 @@ gst_capsfilter_prepare_buf (GstBaseTransform * trans, GstBuffer * input,
         } else {
           ret = GST_FLOW_NOT_NEGOTIATED;
         }
+      } else {
+        gst_capsfilter_push_pending_events (filter, pending_events);
+        pending_events = NULL;
       }
 
       g_list_free_full (pending_events, (GDestroyNotify) gst_event_unref);
-- 
1.8.1.2

